nnsmith/materialize/torch/parse.py

```diff
@@ -0,0 +1,171 @@
+import operator
+from typing import Any, Dict, List, Tuple, Union, cast
+
+import torch
+import torch._dynamo as dynamo
+import torch.fx as fx
+import torch.nn as nn
+import torch.utils._pytree as pytree
+from torch.fx.passes.shape_prop import ShapeProp
+
+from nnsmith.abstract.dtype import DType
+from nnsmith.abstract.op import ConcreteOp, Input
+from nnsmith.abstract.tensor import AbsTensor
+from nnsmith.gir import GraphIR, InstExpr
+from nnsmith.materialize.torch.forward import forward_fn
+
+
+class PropInterpreter(ShapeProp):
+    def run_node(self, n: fx.node.Node) -> Any:
+        result = super().run_node(n)
+        n.meta["res"] = result
+        return result
+
+
+def parse(model: nn.Module, *example_args: List[torch.Tensor]) -> GraphIR:
+    gm: fx.GraphModule = dynamo.export(model, *example_args)[0]
+    # store shape info on nodes
+    sp = PropInterpreter(gm)
+    sp.run(*example_args)
+
+    def load_args(
+        args: Union[List[Any], Dict[str, Any]]
+    ) -> Union[List[Any], Dict[str, Any]]:
+        """
+        map nodes to their outputs while keeping structures and other values the same
```

Comment:
nit: "map" -> "Map"

Reply:
Use https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

Reply:
Yes. The spell checker seems not to suggest capitalization. But generally it is useful to have them to catch potential typos.